packages/authentication/src/web.ts

```diff
@@ -124,6 +129,40 @@ export class FirebaseAuthenticationWeb
     return sendPasswordResetEmail(auth, options.email);
   }
 
+  public async sendSignInLinkToEmail(
+    options: SignInWithEmailLinkOptions,
+  ): Promise<void> {
+    const auth = getAuth();
+    return sendSignInLinkToEmail(
+      auth,
+      options.email,
+      options.actionCodeSettings!,
+    );
+  }
+
+  public isSignInWithEmailLink(
+    options: SignInWithEmailLinkOptions,
+  ): Promise<void> {
+    const auth = getAuth();
+    return new Promise((resolve, reject) => {
```

Comment:
Return the result interface and do not throw an error.

Reply:
I want to change this to your suggestion, but then it complains that the result interface is not a promise. How is this supposed to work that I can return IsSignInWithEmailLinkResult as a promise? Do you have any suggestions?

Reply:
Something like this should work:

```ts
public async isSignInWithEmailLink(
    options: SignInWithEmailLinkOptions,
  ): Promise<void> {
    const auth = getAuth();
    return {
	isSignInWithEmailLink: isSignInWithEmailLink(auth, options.emailLink)
    };
  }
```

Reply:
Ahh, I found the issue. I forgot to add `async` in the function declaration :)